pkg/app.go

```diff
@@ -108,10 +108,27 @@ func (m *Module) handleSerialPayload(data string) {
 		log.Errorln("nil device description found")
 		return
 	}
+	if !devDesc.CheckLength(data) {
+		log.Errorln("invalid payload")
+		return
+	}
+	dataLen := len(data)
+	originalData := data
+	expectedMod := decoder.LoraRawHeaderLen + decoder.LoraRawInnerHeaderLen + decoder.LoraRawCmacLen + decoder.RssiLen + decoder.SnrLen
+	if (dataLen/2)%16 == expectedMod {
+		data = data[14 : dataLen-12]
```

Comment:
change this to be
```
data = data[14 : getInnerPayloadLength(data)]
```
```
func getInnerPayloadLength(data string) int {
    dataLength, _ := strconv.ParseInt(data[12:14], 16, 0)
    return dataLength
}
```
This removes any padding that existed from the encryption